File: backend/app/schemas/multi_agent.py
from typing import Any, Literal

from pydantic import BaseModel, Field


AgentExecutionStatus = Literal["success", "fallback", "failed"]


class MultiAgentAnalyzeRequest(BaseModel):
    market_data_id: int = Field(ge=1)


class MultiAgentAnalyzeResponse(BaseModel):
    workflow_id: str
    workflow_started_at: str
    workflow_completed_at: str
    market_analysis: dict[str, Any]
    risk_analysis: dict[str, Any]
    decision: dict[str, Any]
    explanation: str | None = None
    explanation_status: Literal["llm", "fallback"] | None = None
    explanation_model: str | None = None
    explanation_latency_ms: float | None = None
    explanation_error: str | None = None
    workflow_latency_ms: float | None = None
    agent_latency_ms: dict[str, float] = Field(default_factory=dict)
    agent_status: dict[str, AgentExecutionStatus] = Field(default_factory=dict)
    visited_agents: list[str]
    final_answer: str | None

File: frontend/src/types/multiAgent.ts
export type AgentExecutionStatus =
  | "success"
  | "fallback"
  | "failed";

export interface MarketAnalysisResult {
  node?: string;
  signal?: string;
  [key: string]: unknown;
}

export interface RiskAnalysisResult {
  risk_level?: string;
  risk_score?: number;
  [key: string]: unknown;
}

export interface DecisionResult {
  action?: string;
  [key: string]: unknown;
}

export interface MultiAgentAnalyzeResponse {
  workflow_id: string;
  workflow_started_at: string;
  workflow_completed_at: string;
  market_analysis: MarketAnalysisResult;
  risk_analysis: RiskAnalysisResult;
  decision: DecisionResult;
  explanation: string | null;
  explanation_status: "llm" | "fallback" | null;
  explanation_model: string | null;
  explanation_latency_ms: number | null;
  explanation_error: string | null;
  workflow_latency_ms: number | null;
  agent_latency_ms: Record<string, number>;
  agent_status: Record<string, AgentExecutionStatus>;
  visited_agents: string[];
  final_answer: string | null;
}
